File: src/ds/stack/E_Stack_01.java
package src.ds.stack;

import ds.node.Node;

/**
 * start  10:20
 * end    10:31
 * @author Elena Novichkova
 *
 */

public class E_Stack_01 {
    Node top;

    public boolean isEmpty() {
        return top == null;
    }

    public int peek() {
        if( isEmpty() ) {
            throw new RuntimeException("Stack is empty");
        }
        return top.data;
    }

    public int poll() {
        int data = peek();
        top = top.next;
        return data;
    }

    public void push(int value) {
        Node node = new Node(value);
        node.next = top;
        top = node;
    }
}

File: src/ds/stack/E_Stack_01.js
import Node from "../node/Node.js"

const Stack = function (){
    this.top = null;
    this.isEmpty = function(){
        return this.top == null;
    }
    this.push = function(value){
        let newNode = new Node(value);
        newNode.next = this.top;
        this.top = newNode;
    }
    this.poll = function(){
        let output = this.peek();
        this.top = this.top.next;
        return output;
    }
    this.peek = function(){
        if(this.isEmpty()){
            throw new Error("stack is empty");
        }
        return this.top.data;
    }
}

export default Stack;